starter-code/src/app/services/movies-info.service.ts

```diff
@@ -0,0 +1,119 @@
+    genres: [
+      "Crime", "Drama"
+    ],
+    year: 1994,
+    director: "Frank Darabont",
+    actors: [
+      "Tim Robbins",
+      "Morgan Freeman",
+      "Bob Gunton",
+      "William Sadler",
+      "Clancy Brown",
+      "Gil Bellows"
+    ]
+  }, {
+    id: 2,
+    title: "The Godfather",
+    poster: "https://i.imgur.com/Uzvny9I.jpg",
+    synopsis: "In late summer 1945, guests are gathered for the wedding reception of Don Vito Corleone's daughter Connie (Talia Shire) and Carlo Rizzi (Gianni Russo). Vito...",
+    genres: [
+      "Crime", "Drama"
+    ],
+    year: 1972,
+    director: "Francis Ford Coppola",
+    actors: [
+      "Marlon Brando",
+      "Al Pacino",
+      "James Caan",
+      "Richard S. Castellano",
+      "Robert Duvall",
+      "Sterling Hayden"
+    ]
+  }, {
+    id: 3,
+    title: "The Godfather Part II",
+    poster: "https://i.imgur.com/abJNkWI.jpg",
+    synopsis: "The Godfather Part II presents two parallel storylines. One involves Mafia chief Michael Corleone in 1958/1959 after the events of the first movie; the othe...",
+    genres: [
+      "Crime", "Drama"
+    ],
+    year: 1974,
+    director: "Francis Ford Coppola",
+    actors: [
+      "Al Pacino",
+      "Robert Duvall",
+      "Diane Keaton",
+      "Robert De Niro",
+      "John Cazale",
+      "Talia Shire"
+    ]
+  }, {
+    id: 4,
+    title: "The Dark Knight",
+    poster: "https://i.imgur.com/3jLPB46.jpg",
+    synopsis: "The movie begins with a gang of men with clown masks breaking into the bank where the mob has a large portion of their money stashed. It begins with five cl...",
+    genres: [
+      "Action", "Crime", "Drama", "Thriller"
+    ],
+    year: 2008,
+    director: "Christopher Nolan",
+    actors: [
+      "Christian Bale",
+      "Heath Ledger",
+      "Aaron Eckhart",
+      "Michael Caine",
+      "Maggie Gyllenhaal",
+      "Gary Oldman"
+    ]
+  }, {
+    id: 5,
+    title: "Schindler's List",
+    poster: "https://i.imgur.com/IWZJOmu.jpg",
+    synopsis: "The relocation of Polish Jews from surrounding areas to Krakow begins in late 1939, shortly after the outbreak of World War II, when the German Army defeats...",
+    genres: [
+      "Biography", "Drama", "History"
+    ],
+    year: 1993,
+    director: "Steven Spielberg",
+    actors: [
+      "Liam Neeson",
+      "Ben Kingsley",
+      "Ralph Fiennes",
+      "Caroline Goodall",
+      "Jonathan Sagall",
+      "Embeth Davidtz"
+    ]
+  }
+]
+
+   }
+
+  getMovies() {
+    return this.movies;
+  }
+
+// Por qué no funciona este método getMovie? Me da el siguiente error de compilación:
+// Failed to compile.
+// movies-info.service.ts (112,22): Property 'id' does not exist on type 'Object'.
+  getMovie(id) {
+    let foundMovie = this.movies.findIndex((element) => {
+      return element.id === id;
+    });
+    return this.movies[foundMovie];
```

Comment:
Al principio del service tienes declarado `movies: Object[];` con lo que en este `findIndex((element))` `element` lo mira como `Object`. Cuando pones que es tipo `Object` en typescript, el compilador no sabe que propiedades tiene este objeto, con lo que te dice `Property 'id' does not exist on type 'Object'.`. La solución "sucia" sería al principio del service declarar movies como `movies: any[];` con lo que los elementos serían de tipo any.
La solución "buena" sería crear un modelo `Movie` con las propiedades que necesites y luego declarar movies como ``movies: Movie[];`.